Firebend.AutoCrud.Tests/ChangeTracking/Web/Authorization/EntityChangeTrackingAuthorizationFilterTests.cs

```diff
@@ -28,212 +29,141 @@ public class EntityChangeTrackingAuthorizationFilterTests
+        Assert.ThrowsAsync<ArgumentException>(() =>
+            entityChangeTrackingAuthorizationFilter.OnActionExecutionAsync(_actionExecutingContext.Object,
+                _nextDelegate.Object));
+        _nextDelegate.Verify(x => x(), Times.Never);
     }
 
     [Test]
-    public void Should_Next_If_Key_Parser_Fails_To_Parse()
+    public async Task Should_Return_403_If_Id_Is_Not_Null_And_Authorization_Fail()
     {
         // given
-        var entityReadService = new Mock<IEntityReadService<Guid, EntityChangeTrackingAuthorizationTestClass>>();
-        _serviceProvider.Setup(s
-            => s.GetService(typeof(IEntityReadService<Guid, EntityChangeTrackingAuthorizationTestClass>))).Returns(entityReadService.Object);
-
-        var entityKeyParser = new Mock<IEntityKeyParser<Guid, EntityChangeTrackingAuthorizationTestClass>>();
-        _serviceProvider.Setup(s
-            => s.GetService(typeof(IEntityKeyParser<Guid, EntityChangeTrackingAuthorizationTestClass>))).Returns(entityKeyParser.Object);
-
-        var authorizationService = new Mock<IAuthorizationService>();
-        authorizationService.Setup(a => a.AuthorizeAsync(
+        _entityAuthProvider.Setup(a => a.AuthorizeEntityAsync<Guid, ActionFilterTestHelper.TestEntity>(
+            It.IsAny<string>(),
             It.IsAny<ClaimsPrincipal>(),
-            It.IsAny<object>(),
-            It.IsAny<string>()
-        )).ReturnsAsync(AuthorizationResult.Failed(AuthorizationFailure.Failed(new[] { new ChangeTrackingAuthorizationRequirement() })));
-        _serviceProvider.Setup(s =>
-            s.GetService(typeof(IAuthorizationService))).Returns(authorizationService.Object);
+            It.IsAny<string>(),
+            It.IsAny<CancellationToken>()
+        )).ReturnsAsync(
+            AuthorizationResult.Failed(
+                AuthorizationFailure.Failed(new[] { new ChangeTrackingAuthorizationRequirement() })));
 
-        var actionExecutingContext = _fixture.Create<Mock<ActionExecutingContext>>();
-
-        var actionArguments = new Dictionary<string, object> { { "EntityId", Guid.NewGuid().ToString() } };
-        actionExecutingContext.Setup(a => a.ActionArguments).Returns(actionArguments);
+        var actionArguments = new Dictionary<string, object> { { "entityId", Guid.NewGuid().ToString() } };
+        _actionExecutingContext.Setup(a => a.ActionArguments).Returns(actionArguments);
 
         // when
         var entityChangeTrackingAuthorizationFilter =
-            new EntityChangeTrackingAuthorizationFilter<Guid, EntityChangeTrackingAuthorizationTestClass>(_policy);
+            new EntityChangeTrackingAuthorizationFilter<Guid, ActionFilterTestHelper.TestEntity>(_policy);
 
         // then
-        entityChangeTrackingAuthorizationFilter.OnActionExecutionAsync(actionExecutingContext.Object, Next);
+        await entityChangeTrackingAuthorizationFilter.OnActionExecutionAsync(_actionExecutingContext.Object,
+            _nextDelegate.Object);
```

Comment:
Not a huge deal, but it seems like this belongs in the "when" section instead of "then"